designs/FloatingStableAndPrerelease.md

```diff
@@ -0,0 +1,95 @@
+I should be able to use both the braces/regex formats i.e. 1.* or [1.0.0, 2.0.0) that can include pre-release versions
+I would like to additionally define whether I want to resolve to rc, beta, alpha or all pre-releases.
+
+## Goals
+
+* Define a versioning strategy that allows floating to both stable and prerelease versions simultaneously. Floating only prerelease and only stable versions is possible today with `1.0.0-*` and `1.*` respectively.
+
+## Non-Goals
+
+* Support for more than 1 prerelease label in floating versions. This is out of scope, because there is no way today with the prerelease only floating versions. As such it can be treated as a separate feature.
+
+## Solution
+
+The proposed syntax is one commonly discussed in the linked issue, `*-*`.
+Specifically `*-*` says, I want the absolute latest and greatest package, and it doesn't matter to me whether the latest version is stable or prerelease.
+Notable that this approach merely builds on top of the current floating version approach. The stable part syntax is the same and the prerelease syntax stays the same. The difference is, we can now combine them.
+For example whenever we float a minor and prerelease, ex `1.*-*`, the stable part can match individually, but the prerelease part has to have a matching stable part. 
+
+For example:
+
+| Floating Version | Available Versions | Versions that match |
+| -----------------|--------------------|-----------------------|
+| 5.*-rc.* | 4.9.0 <br> 5.1.0 <br> 5.2.0-beta.1 <br> 5.2.0-rc.3 <br> 6.0.0-rc.1 | 5.1.0 <br> 5.2.0-rc.3 |
+
+Note that 5.2.0-beta.1 and 6.0.0-rc.1 do not match.
+
+The behavior is best illusrated with examples
+
+| Floating Version | Available Versions | Matching Versions | Best version | Notes |
+| -----------------|--------------------|-------------------|--------------|-------|
+| *                | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 <br> 3.0.0-beta.1 | 1.1.0 <br> 2.0.0 | 2.0.0 | Only stable versions are matched | 
+| 1.*              | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 <br> 3.0.0-beta.1 | 1.1.0 | 1.1.0 | Only stable versions starting 1.* are matched | 
+| 1.2.0-*          | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 <br> 3.0.0-beta.1 | 1.2.0-rc.1 <br> 1.2.0-rc.2 | 1.2.0-rc.2 | The highest matching prerelease is selected |
+| *-*              | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 <br> 3.0.0-beta.1 | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 <br> 3.0.0-beta.1 | 3.0.0-beta.1 | 3.0.0 is the highest version |
+| 1.*-*              | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 <br> 3.0.0-beta.1 | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 | 1.2.0-rc.2 | A prerelease version is the highest matching version |
+| *-rc.*              | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 <br> 3.0.0-beta.1 | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 | 2.0.0 | A stable version is the highest matching version |

```

Comment:
please include a sample where the star is not immediately after a dot.

`1.0.0-rc*`
and
`1.0.0-rc.1*`

Reply:
fixed.